packages/svelte/src/compiler/phases/2-analyze/visitors/shared/component.js

```diff
@@ -15,6 +16,70 @@ import { mark_subtree_dynamic } from './fragment.js';
  * @param {Context} context
  */
 export function visit_component(node, context) {
+	node.metadata.path = [...context.path];
+
+	// link this node to all the snippets that it could render, so that we can prune CSS correctly
+	node.metadata.snippets = new Set();
+
+	let resolved = true;
+
+	for (const attribute of node.attributes) {
```

Comment:
probably not really relevant, but this iterates over the attributes, and below we're iterating over them again, feels a bit wasteful - but merging them into one probably makes everything harder to reason about